rocAL/source/readers/image/coco_file_source_reader.cpp

```diff
@@ -197,30 +224,40 @@ int COCOFileSourceReader::release() {
 
 void COCOFileSourceReader::shuffle_with_aspect_ratios() {
     // Calculate the mid element which divides the aspect ratios into two groups (<=1.0 and >1.0)
-    auto mid = std::upper_bound(_aspect_ratios.begin(), _aspect_ratios.end(), 1.0f) - _aspect_ratios.begin();
-    // Shuffle within groups using the mid element as the limit - [start, mid) and [mid, last)
-    std::random_shuffle(_file_names.begin(), _file_names.begin() + mid);
-    std::random_shuffle(_file_names.begin() + mid, _file_names.end());
+    auto mid = std::upper_bound(_aspect_ratios.begin() + get_start_idx(), _aspect_ratios.begin() + get_start_idx() + shard_size_without_padding(), 1.0f) - (_aspect_ratios.begin() + get_start_idx());
```

Comment:
Get the start index once, instead of computing multiple times here in this function